app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -842,7 +842,14 @@ private void moveOverlayUpOrDownToMatchAppraisalBox() {
             return;
         }
 
-        moveOverlay(appraisalBox.getVisibility() == View.VISIBLE);
+        if (inputBox.getVisibility() == View.VISIBLE) {
+            boolean appraisalExpanded = appraisalBox.getVisibility() == View.VISIBLE;
+            moveOverlay(appraisalExpanded);
+        } else { //always move overlay down if not on input screen. Appraisal box cant be open if user is not in
+            // input screen
+            moveOverlay(false);
+        }
+
```

Comment:
I didn't mean how the program ran the check, I thought you meant that I should add moveOverlay(inputBox.getVisibility() == View.VISIBLE && appraisalBox.getVisibility() == View.VISIBLE); to replace line 850, which wouldn't help much, which I explained by referring to that the possibility would never be possible. 

Do you mean that the if-else block should be replaced by 
moveOverlay(inputBox.getVisibility() == View.VISIBLE && appraisalBox.getVisibility() == View.VISIBLE);?

In that case, the answer is yes, I think it could be.